src/DataStructures/VectorImpl.hpp

```diff
@@ -210,7 +211,10 @@ class VectorImpl
             &free};
         reset_pointer_vector(new_size);
       } else {
-        ERROR("may not destructively resize a non-owning vector");
+        ERROR(MakeString{}
```

Comment:
why is this an ERROR and not an ASSERT?

Reply:
I think the reasoning here is that if this function returns, it should guarantee that the vector has the new size, and it genuinely can't perform the resize in this branch. You could argue that it could just leave the vector size be, but I think we need to evaluate the `if` regardless, and replacing the error with an `ASSERT` followed by a return probably just ensures that a release build will segfault instead of erroring...

Reply:
Nevermind, sorry, I was thinking about this badly. Yes, we can just `ASSERT(owning_, [helpful error message])` above and not evaluate the branch.